ops/pebble.py

```diff
@@ -1189,6 +1247,22 @@ def __eq__(self, other: Union['LogTargetDict', 'LogTarget']):
         else:
             return NotImplemented
 
+    def _merge(self, other: 'LogTarget'):
+        """Merges this log target object with another log target definition.
+
+        For attributes present in both objects, the passed in log target
+        attributes take precedence.
+        """
+        for name, value in other.__dict__.items():
+            if not value or name == 'name':
+                continue
+            if name == 'labels':
+                getattr(self, name).update(value)
```

Comment:
`self.labels` is actually an optional dict, but I think it should be a non-optional dict, to make this a bit simpler. I think we can just make that change. In `LogTarget.__init__`, could you please make this change? Change this:

```python
        if labels is not None:
            labels = copy.deepcopy(labels)
        self.labels: Optional[Dict[str, str]] = labels
```

to this:

```python
        if labels is not None:
            labels = copy.deepcopy(labels)
        else:
            labels = {}
        self.labels: Dict[str, str] = labels
```

You'll have to update an assert in `test_pebble.py` to `assert target.labels == {}` too.